sources/packages/api/src/services/designation-agreement/designation-agreement.service.ts

```diff
@@ -96,4 +96,49 @@ export class DesignationAgreementService extends RecordDataModelService<Designat
       .andWhere("location.institution.id = :institutionId", { institutionId })
       .getOne();
   }
+
+  /**
+   * Get the summary of all designation agreements for the institution.
+   * @param institutionId institution to retrieve the designations.
+   * @returns designations under the institution.
+   */
+  async getInstitutionDesignationsById(
+    institutionId: number,
+  ): Promise<DesignationAgreement[]> {
+    return this.repo
+      .createQueryBuilder("designation")
+      .select([
+        "designation.id",
+        "designation.designationStatus",
+        "designation.submittedDate",
+        "designation.startDate",
+        "designation.endDate",
+      ])
+      .where("designation.institution.id = :institutionId", { institutionId })
+      .orderBy(
+        `CASE designation.designationStatus
+            WHEN '${DesignationAgreementStatus.Pending}' THEN 1
+            WHEN '${DesignationAgreementStatus.Approved}' THEN 2
+            WHEN '${DesignationAgreementStatus.Declined}' THEN 3
+            ELSE 4
+          END`,
```

Comment:
Is it possible for 1, 2, 3 and 4 to be Enums so that it will a bit more meaningful ?

Reply:
In this case the numbers are supposed to be really a number that will be used to imposed the proper order to the query results. What the enum would look like? Something like below? @guru-aot, @ann-aot, @dheepak-aot, do you believe that it would be readable?
```javascript
enum SortPriority {
    Priority1 = 1,
    Priority2 = 2,
    Priority3 = 3,
    Priority4 = 4,

WHEN '${DesignationAgreementStatus.Declined}' THEN ${SortPriority.Priority1}
}
```

Reply:
this looks fine to me

Reply:
I updated the code per the enum that I shared above. Please let me know if it is ok @ann-aot, @henryoforeh-aot, @dheepak-aot, @guru-aot.

Reply:
Sorry for being late to reply on this. 
IMO as per enum, when we declare enum as 

enum SortPriority {
    Priority1,
    Priority2,
    Priority3,
    Priority4,
}

 the value is automatically set as 1,2,3,4 for Priority1 .. Priority4 respectively. 

Also as we know the enum is for Priority we can declare like

enum SortPriority {
    ONE,
    TWO,
    THREE,
    FOUR,
}

Here comes another thought. If the order of enum items declared in DB is same as their priority  then we can avoid this WHEN and THEN. I know it won't be possible always with all the enums. So in exceptional cases SortPriority enum can be used,

Reply:
I got the idea about ONE, TWO, THREE, FOUR but to me it is going to far 😄 
To avoid using numbers we are converting numbers to its written format. It does not make much sense to me.
Yes, we should avoid "magic numbers" in the code, but is it the case for this specific one?

Reply:
ONE, TWO, THREE, FOUR is just a convention which came to my mind. Naming is up to the dev to decide. But my main opinion is not assigning numbers. to enum and leverage it's own benefit.